modules/reindex/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecReindexIT.java

```diff
@@ -61,72 +57,94 @@ public void testReindexingMultipleCodecs() throws InterruptedException, Executio
             "lz4",
             "BEST_SPEED"
         );
-
-        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
-            assertReindexingWithMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
-        }
-
-    }
-
-    private void assertReindexingWithMultipleCodecs(String destCodec, String destCodecMode, Map<String, String> codecMap)
-        throws ExecutionException, InterruptedException {
-
-        final String index = "test-index" + destCodec;
-        final String destIndex = "dest-index" + destCodec;
-
-        // creating source index
-        createIndex(
-            index,
-            Settings.builder()
-                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
-                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
-                .put("index.codec", "default")
-                .put("index.merge.policy.max_merged_segment", "1b")
-                .build()
-        );
-        ensureGreen(index);
-
         final int nbDocs = randomIntBetween(2, 5);
+        final String index = "source-index";
+
+        {  // Setup
+           // create source index
+            createIndex(
+                index,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put("index.codec", "default")
+                    .put("index.merge.policy.max_merged_segment", "1b")
+                    .build()
+            );
+            ensureGreen(index);
+
+            // index using all codecs
+            for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+                useCodec(index, codec.getKey());
+
+                // perform index
+                indexRandom(
+                    randomBoolean(),
+                    false,
+                    randomBoolean(),
+                    IntStream.range(0, nbDocs)
+                        .mapToObj(i -> client().prepareIndex(index).setId(UUID.randomUUID().toString()).setSource("num", i))
+                        .collect(toList())
+                );
+
+                // perform flush
+                FlushResponse flushResponse = client().admin().indices().prepareFlush(index).setForce(true).execute().actionGet();
+                assertNoFailures(flushResponse);
```

Comment:
Maybe replace with: https://github.com/opensearch-project/OpenSearch/blob/73bbeb57a8a56de0d1f9cf69e711a8a0ff26afe5/test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java#L1456

```suggestion
                flushAndRefresh(index);
```

Reply:
yes, please replace it with the flushAndRefresh function